doc/rst/developer/bestPractices/TestSystem.rst

```diff
@@ -466,6 +638,182 @@ include a future for the issue tracked on the `GitHub issues page`_.
+foo.noexec          empty file. Indicates .chpl file should only be compiled,
+                    not executed.  See `Controlling How It Runs`_ for more
+                    information.
+foo.notest          empty file. Indicates the file should not be run explicitly
+                    See `Controlling How It Runs`_ for more information.
+NOTEST              empty file. Indicates the directory should not be run
+foo.skipif          line separated list of conditions under which the test
+                    should not be run, or a script to compute the same.  See
+                    `Limiting Where the Test Runs`_ for more information
+SKIPIF              same as above, but applied to the entire directory
+foo.suppressif      line separated list of conditions under which the test is
+                    expected to fail, or a script to compute the same
+foo.timeout         time in seconds after which start_test should stop this test
+                    See `Limiting Time Taken`_ for more information
+..
+-------------------------------------------------------------------------------
+**performance**
+-------------------------------------------------------------------------------
+foo.perfcompopts    compiler flags, overrides .compopts for --performance
+PERFCOMPOPTS        directory-wide performance compiler flags
+foo.perfexecopts    runtime flags, overrides .execopts for --performance
+PERFEXECOPTS        directory-wide performance runtime flags
+foo.perfexecenv     environment variables, overrides .execenv for --performance
+PERFEXECENV         directory-wide performance environment variables
+foo.perfnumtrials   number of execution trials to run if no timeout specified
+PERFNUMTRIALS       directory-wide number of execution trials to run
+foo.perftimeout     time in seconds after which start_test should stop this test
+foo.perfkeys        keys to search for in the output
+foo.graph           Specifies which data files and perfkeys to graph, and
+                    contains meta-data associated with labeling data sets,
+                    axis, and graphs
+test/GRAPHFILES     Acts as an index that tracks all .graph that should be
+                    graphed.
+..
+-------------------------------------------------------------------------------
+**futures**
+-------------------------------------------------------------------------------
+foo.future          Describes the future being tested, following the
+                    newline-separated format of:
+                    *category*, *title*, *issue #*
+foo.bad             output generated on a failing test, to track if a known
+                    failing future begins failing a different way.  See
+                    `Tracking Current Failure Mode`_ for more information
+..
+=================   ===========================================================
+
 .. _extensions:
 
 Planned Changes of Testing System
```

Comment:
You haven't changed this, but I would just remove this section, or at least make it a non-rendered section

We're not going to get to YAML anytime soon (and if we ever do this document will change drastically), tracking of reference benchmarks isn't that relevant to external devs, and I think this PR improves documentation enough that the last bullet can just be removed.